pkg/cmd/project/item-add/item_add.go

```diff
@@ -0,0 +1,157 @@
+gh project item-add 1 --org github --url https://github.com/cli/go-gh/issues/1
+
+# add --format=json to output in JSON format
+`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				opts.number, err = strconv.Atoi(args[0])
+				if err != nil {
+					return fmt.Errorf("invalid number: %v", args[0])
+				}
+			}
+
+			t := tableprinter.New(f.IOStreams)
+			config := addItemConfig{
+				tp:     t,
+				client: client,
+				opts:   opts,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runAddItem(config)
+		},
+	}
+
+	addItemCmd.Flags().StringVar(&opts.userOwner, "user", "", "Login of the user owner. Use \"@me\" for the current user.")
+	addItemCmd.Flags().StringVar(&opts.orgOwner, "org", "", "Login of the organization owner.")
+	addItemCmd.Flags().StringVar(&opts.itemURL, "url", "", "URL of the issue or pull request to add to the project. Note that the name of the owner is case sensitive, and will fail to find the item if it does not match. Must be of form https://github.com/OWNER/REPO/issues/NUMBER or https://github.com/OWNER/REPO/pull/NUMBER")
+	addItemCmd.MarkFlagsMutuallyExclusive("user", "org")
+	addItemCmd.Flags().StringVar(&opts.format, "format", "", "Output format, must be 'json'.")
+
+	_ = addItemCmd.MarkFlagRequired("url")
+
+	return addItemCmd
+}
+
+func runAddItem(config addItemConfig) error {
+	if config.opts.format != "" && config.opts.format != "json" {
+		return fmt.Errorf("format must be 'json'")
+	}
+
+	owner, err := queries.NewOwner(config.client, config.opts.userOwner, config.opts.orgOwner)
+	if err != nil {
+		return err
+	}
+
+	project, err := queries.NewProject(config.client, owner, config.opts.number, false)
+	if err != nil {
+		return err
+	}
+	config.opts.projectID = project.ID
+
+	itemID, err := queries.IssueOrPullRequestID(config.client, config.opts.itemURL)
+	if err != nil {
+		return err
+	}
+
+	config.opts.itemID = itemID
+
+	query, variables := addItemArgs(config)
+	err = config.client.Mutate("AddItem", query, variables)
+	if err != nil {
+		return err
+	}
+
+	if config.opts.format == "json" {
+		return printJSON(config, query.CreateProjectItem.ProjectV2Item)
+	}
+
+	return printResults(config, query.CreateProjectItem.ProjectV2Item)
+
+}
+
+func addItemArgs(config addItemConfig) (*addProjectItemMutation, map[string]interface{}) {
+	return &addProjectItemMutation{}, map[string]interface{}{
+		"input": githubv4.AddProjectV2ItemByIdInput{
+			ProjectID: githubv4.ID(config.opts.projectID),
+			ContentID: githubv4.ID(config.opts.itemID),
+		},
+	}
+}
+
+func printResults(config addItemConfig, item queries.ProjectItem) error {
+	// using table printer here for consistency in case it ends up being needed in the future
+	config.tp.AddField("Added item")
```

Comment:
Ditto: not convinced that tableprinter is a good mechanism to print anything other than structured data that spans multiple columns.

Reply:
what would you suggest changing it to?